src/pybind/mgr/rbd_support/module.py

```diff
@@ -0,0 +1,498 @@
+                sum_namespace = sum_namespaces.setdefault(namespace, {})
+                for image_id, raw_image in raw_images.items():
+                    # zero-out non-updated raw counters
+                    if not raw_image[0]:
+                        continue
+                    elif raw_image[0][0] < now_ts:
+                        raw_image[1] = raw_image[0]
+                        raw_image[0] = [now_ts, [0 for x in raw_image[1][1]]]
+                        continue
+
+                    counters = raw_image[0][1]
+
+                    # copy raw counters if this is a newly discovered image or
+                    # increment existing counters
+                    sum_image = sum_namespace.setdefault(image_id, None)
+                    if sum_image:
+                        for i in range(len(counters)):
+                            sum_image[i] += counters[i]
+                    else:
+                        sum_namespace[image_id] = [x for x in counters]
+
+        self.log.debug("sum_osd_perf_counters: {}".format(sum_pool_counters))
+        return sum_pool_counters
+
+    def refresh_image_names(self, resolve_image_names):
+        rbd = RBD()
+        for pool_id, namespace in resolve_image_names:
+            image_key = (pool_id, namespace)
+            images = self.image_name_cache.setdefault(image_key, {})
+            with self.rados.open_ioctx2(int(pool_id)) as ioctx:
+                ioctx.set_namespace(namespace)
+                for image_meta in RBD().list2(ioctx):
+                    images[image_meta['id']] = image_meta['name']
+            self.log.debug("resolve_image_names: {}={}".format(image_key, images))
+
+    def scrub_missing_images(self):
+        for pool_key, query in self.user_queries.items():
+            raw_pool_counters = query.get(QUERY_RAW_POOL_COUNTERS, {})
+            sum_pool_counters = query.get(QUERY_SUM_POOL_COUNTERS, {})
+            for pool_id, sum_namespaces in sum_pool_counters.items():
+                raw_namespaces = raw_pool_counters.get(pool_id, {})
+                for namespace, sum_images in sum_namespaces.items():
+                    raw_images = raw_namespaces.get(namespace, {})
+
+                    image_key = (pool_id, namespace)
+                    image_names = self.image_name_cache.get(image_key, {})
+                    for image_id in list(sum_images.keys()):
+                        # scrub image counters if we failed to resolve image name
+                        if image_id not in image_names:
+                            self.log.debug("scrub_missing_images: dropping {}/{}".format(
+                                image_key, image_id))
+                            del sum_images[image_id]
+                            if image_id in raw_images:
+                                del raw_images[image_id]
+
+    def process_raw_osd_perf_counters(self):
+        now = datetime.now()
+        now_ts = int(now.timestamp())
```

Comment:
for python2:
```
2019-01-28 13:09:36.265 7ff68dd60700  0 mgr[rbd_support] Fatal runtime error: 'datetime.datetime' object has no attribute 'timestamp'
Traceback (most recent call last):
  File "/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py", line 137, in run
    self.process_raw_osd_perf_counters()
  File "/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py", line 255, in process_raw_osd_perf_counters
    now_ts = int(now.timestamp())
AttributeError: 'datetime.datetime' object has no attribute 'timestamp'
```

Reply:
`now_ts = int(now.strftime("%s"))` could be used.

Reply:
Thx -- updated